stream.go

```diff
@@ -0,0 +1,55 @@
+package sm_yamux
+
+import (
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/mux"
+	"github.com/libp2p/go-yamux"
+)
+
+// stream implements mux.MuxedStream over yamux.Stream.
+type stream yamux.Stream
+
+func (s *stream) Read(b []byte) (n int, err error) {
+	n, err = s.yamux().Read(b)
+	if err == yamux.ErrConnectionReset {
+		err = mux.ErrReset
+	}
+
+	return
+}
+
+func (s *stream) Write(b []byte) (n int, err error) {
+	n, err = s.yamux().Write(b)
+	if err == yamux.ErrConnectionReset {
```

Comment:
Check ErrStreamClosed?

Reply:
Don't think that it should return `ErrReset` on `ErrStreamClosed`, cause that is a different error. I've suggested introducing an additional error for writing on a closed stream, but I also have a concern about it described here libp2p/go-libp2p#814.

Reply:
I could have sworn I saw an `ErrClosed` error in defined. Nevermind.